pkg/controllers/managementuser/rbac/namespace_handler.go

```diff
@@ -490,6 +493,22 @@ func addManageNSPermission(clusterRole *rbacv1.ClusterRole, projectName string)
 	return clusterRole
 }
 
+func addUpdatePSAPermission(cr *rbacv1.ClusterRole, projectName string) *rbacv1.ClusterRole {
+	if cr.Rules == nil {
+		cr.Rules = []rbacv1.PolicyRule{}
+	}
+	cr.Rules = append(cr.Rules, rbacv1.PolicyRule{
+		APIGroups:     []string{management.GroupName},
+		Verbs:         []string{updatePSAVerb},
+		Resources:     []string{v32.ProjectResourceName},
+		ResourceNames: []string{projectName},
+	})
+	if cr.Annotations == nil {
```

Comment:
why do we need this?

Reply:
I took inspiration from this function already present on the code base: https://github.com/rancher/rancher/blob/e66f0ca975a8e59d556bb2cd6cd72eb9168bef2b/pkg/controllers/managementuser/rbac/namespace_handler.go#L477

Reply:
I'm not sure why that was introduced, but unless I'm missing something it looks to me that this is not needed.